WordPress/gutenberg
# Fix the headings hierarchy in the styles sidebar

Fixes #42496 

## What?

Changes the heading levels in the styles sidebar so that the headings hierarchy is correct.

## Why?

The heading hierarchy must reflect the logical structure of the content.

Previously, the first heading in the styles sidebar was a H5, which skipped many levels and didn't make sense.

Also, sections of content that belong to a main 'topic' need a heading level to indicate they're 'sub-headings'.

## How?

- Changes the level of some `Heading` components.
- Chances the heading level of some other components that use a `Heading` component, adds new props where necessary.

Regarding `ToolsPanel`, I'm not that familiar with TypeScript types, please do let me know if the way I imported and used types is correct.

## Testing Instructions




The best way to test this PR is by installing the [HeadingsMap](https://chrome.google.com/webstore/detail/headingsmap/flbjommegcjonpdmenkdiocclhjacmbi) browser extension (available also for Firefox).
- Go to the site editor.
- Open the Styles sidebar.
- Launch HeadingsMap: it will open a sidebar that shows the headings hierarchy.
- Navigate through **all the panels and sub-panels** of the Styles sidebar: the HeadingsMap will update on the fly and show the headings hierarchy of the current panel.
- Check that the headings hierarchy within the Styles sidebar:
  - starts with a H2
  - all the headings below the H2 are sub-headings (H3 and H4, depending on the panel)


## Screenshots or screencast 

A few screenshots:

Typography panel:

<img width="1792" alt="typography" src="https://user-images.githubusercontent.com/1682452/188426098-5db0abb5-174f-4705-9dc0-e1772b553c7a.png">

Colors panel:

<img width="1792" alt="colors" src="https://user-images.githubusercontent.com/1682452/188426225-70843b65-947d-45e6-bbec-6e016931b7c4.png">

Colors > Palette panel:

<img width="1792" alt="color palette" src="https://user-images.githubusercontent.com/1682452/188426508-7324efd7-1bc4-4b86-97a6-88bae2a76c00.png">

Layout panel:

<img width="1791" alt="layout" src="https://user-images.githubusercontent.com/1682452/188426579-da6f2676-e322-4814-b382-3755cb7fbee0.png">
